packages/react-components/src/AzeroId.tsx

```diff
@@ -0,0 +1,185 @@
+import type { ApiPromise } from '@polkadot/api';
+
+import { SupportedChainId } from '@azns/resolver-core';
+import { useResolveAddressToDomain } from '@azns/resolver-react';
+import React, { useCallback, useId } from 'react';
+import CopyToClipboard from 'react-copy-to-clipboard';
+
+import { externalAzeroIdLogoBlackSVG, externalAzeroIdLogoGreySVG, externalAzeroIdLogoPrimarySVG } from '@polkadot/apps-config/ui/logos/external';
+import { useApi, useQueue, useTheme } from '@polkadot/react-hooks';
+
+import Icon from './Icon.js';
+import { styled } from './styled.js';
+import Tooltip from './Tooltip.js';
+import { useTranslation } from './translate.js';
+
+type WrappedAzeroIdProps = {
+  address?: string;
+  className?: string;
+  isRegisterLinkShown?: boolean;
+};
+
+type AzeroIdProps = WrappedAzeroIdProps & {
+  api: ApiPromise;
+  chainId: SupportedChainId.AlephZero | SupportedChainId.AlephZeroTestnet,
+};
+
+export const systemNameToChainId: Map<string, SupportedChainId.AlephZero | SupportedChainId.AlephZeroTestnet> = new Map([
+  ['Aleph Zero', SupportedChainId.AlephZero],
+  ['Aleph Zero Testnet', SupportedChainId.AlephZeroTestnet]
+]);
```

Comment:
Is this map ever going to have to be modifiable? If not (i.e. if it's a _const_ const) then I don't think we gain anything from it being a `Map` and it'd be simpler (and even more type-proper) to make it `as const` object:
```suggestion
export const SYSTEM_NAME_TO_CHAIN_ID = {
  'Aleph Zero': SupportedChainId.AlephZero,
  'Aleph Zero Testnet': SupportedChainId.AlephZeroTestnet,
} as const
```

Reply:
Map helps me with typing. In multiple places I do `systemNameToChainId[systemChain]` where `systemChain` any string.
With Map it works and returns `string | undefined`. With const it throws because `AlephZero` != `string`. I'd keep it for this reason, unless there is a way to solve it

Reply:
Oh ok, so maybe indeed an explicit type is required like you did. But then I still think it doesn't need to be a `Map` and can be regular object? I mean, the _api_ of a regular object is simpler :P
```suggestion
export const SYSTEM_NAME_TO_CHAIN_ID: { [key: string]: SupportedChainId } = {
  'Aleph Zero': SupportedChainId.AlephZero,
  'Aleph Zero Testnet': SupportedChainId.AlephZeroTestnet,
} as const
```

Just a suggestion.

Reply:
but then I get `string` instead of `string | undefined` for any string :sweat_smile: 
It fees like Map is safer. Or I can create some getter function that would take any string and return `string | undefined` which will work like Map.get :shrug:

Reply:
Hmm, that's true - I think it depends on ts config, but not that I'd get into that.